src/atoms/Text/Text.web.js

```diff
@@ -1,3 +1,122 @@
-const Text = () => null;
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import baseTheme from '../../tokens/theme';
+import { getColorKeys, getColor, getLineHeight } from '../../_helpers/theme';
+
+const styles = {
+  fontFamily({ theme, _weight }) {
+    return theme.fonts.family.lato[_weight];
+  },
+  fontSize({ theme, size }) {
+    return theme.fonts.size[size];
+  },
+  color({ theme, color }) {
+    return getColor(theme, color);
+  },
+  textDecorationLine({ _isUnderlined }) {
+    if (_isUnderlined) {
+      return 'underline';
+    } else {
+      return 'none';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ theme, _letterSpacing }) {
+    return `${theme.fonts.letterSpacing[_letterSpacing]}`;
+  },
+  lineHeight({ theme, size, _lineHeight }) {
+    if (_lineHeight) {
+      return theme.fonts.lineHeight[_lineHeight];
+    } else {
+      return getLineHeight(theme, size);
+    }
+  },
+  overflow({ truncate }) {
+    if (truncate) {
+      return 'hidden';
+    } else {
+      return '';
+    }
+  },
+  whiteSpace({ truncate }) {
+    if (truncate) {
+      return 'nowrap';
+    } else {
+      return '';
+    }
+  },
+  textOverflow({ truncate }) {
+    if (truncate) {
+      return 'ellipsis';
+    } else {
+      return '';
+    }
+  },
+};
+
+const Text = styled(
+  ({
+    as,
+    fontFamily,
+    fontSize,
+    children,
+    textDecorationLine,
+    align,
+    letterSpacing,
+    lineHeight,
+    style,
+    ...props
+  }) =>
+    React.createElement(as, props, children, {
```

Comment:
```as``` is a styled-components prop which when passed will use it and your code will never reach this createElement line.

https://styled-components.com/docs/api#as-polymorphic-prop

You can just do 
```
const Text = styled.div`
```

and everything will work

Reply:
wont it restrict to be only div ?

Reply:
It wont. It will be ```div``` if no ```as``` is passed. If you pass ```as```  then it will take that as the component

Reply:
Done